### PR TITLE
Add psbt_signer.rs example

### PR DESCRIPTION
**Pull request by notmandatory**
### Description

Adding a simple example of how to create a PSBT with a watch only wallet and then sign it with a signing wallet. 

### Notes to the reviewers

This example was inspired by a question from a user.

### Changelog notice

none.

#### All Submissions:

* [x] I've signed all my commits
* [x] I followed the [contribution guidelines](https://github.com/bitcoindevkit/bdk/blob/master/CONTRIBUTING.md)
* [x] I ran `cargo fmt` and `cargo clippy` before committing

#### New Features:

* [ ] I've added tests for the new feature
* [ ] I've added docs for the new feature

#### Bugfixes:

* [ ] This pull request breaks the existing API
* [ ] I've added tests to reproduce the issue which are now passing
* [ ] I'm linking the issue being fixed by this PR

### CODE REVIEW

**Comment by notmandatory** [comment]
@vladimirfomene please take a look and see if it works for you.

**Review by vladimirfomene — COMMENTED** [review]
Tested ACK

**Review by vladimirfomene — COMMENTED** [review]
Tested ACK

File: examples/psbt_signer.rs [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    )?;
+
+    #[cfg(feature = "electrum")]
+    {
+        println!("Sync watch only wallet");
+        watch_only_wallet.sync(&blockchain, SyncOptions::default())?;
+    }
+
+    // get deposit address
+    let deposit_address = watch_only_wallet.get_address(AddressIndex::New).unwrap();
```
Comment (danielabrozzoni) [inline examples/psbt_signer.rs]: nit: you should probably `?` here, instead of unwrapping

File: examples/psbt_signer.rs [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    let balance = watch_only_wallet.get_balance()?;
+    println!("Watch only wallet balance in SAT: {}", balance);
+
+    //Create the PSBT sending 10,000 SATs back to the same wallet
+    let send_to_address = watch_only_wallet.get_address(AddressIndex::New)?;
+    let mut builder = watch_only_wallet.build_tx();
+    builder
+        .add_recipient(send_to_address.script_pubkey(), 10_000)
+        .enable_rbf()
+        .do_not_spend_change()
```
Comment (danielabrozzoni) [inline examples/psbt_signer.rs]: Do we need this? This adds all the change outputs to the "unspendable" list (see https://docs.rs/bdk/latest/bdk/wallet/tx_builder/struct.TxBuilder.html#method.do_not_spend_change), I fear that adding it in a example without a real reason for doing so might confuse users, and they might just copy it and then wonder why they can't spend most of their funds...

File: examples/psbt_signer.rs [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    }
+
+    // get deposit address
+    let deposit_address = watch_only_wallet.get_address(AddressIndex::New).unwrap();
+    println!(
+        "Fund watch only wallet by sending testnet bitcoin to this address: {addr}\nExplorer URL: https://mempool.space/testnet/address/{addr}",
+        addr = deposit_address.address
+    );
+
+    let balance = watch_only_wallet.get_balance()?;
```
Comment (danielabrozzoni) [inline examples/psbt_signer.rs]: I would modify this code a bit, to make the example easier to use:

First, check the balance, if it's less than ~10_200 sats (you're sending 10_000 sats, I'm adding a 200 sats margin for fees), print the message asking the user to fund the wallet, and exit. Otherwise, continue with creating the transaction.

File: examples/psbt_signer.rs [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    println!("Watch only wallet balance in SAT: {}", balance);
+
+    //Create the PSBT sending 10,000 SATs back to the same wallet
+    let send_to_address = watch_only_wallet.get_address(AddressIndex::New)?;
```
Comment (danielabrozzoni) [inline examples/psbt_signer.rs]: nit: maybe print the send_to_address in as well, so that we know how to recover the funds :) Otherwise, we might just want to hardcode here a testnet faucet address.

**Review by danielabrozzoni — COMMENTED** [review]
Concept ACK

It looks good to me, I think it should also be added to the `Cargo.toml` in the examples section :)

File: examples/psbt_signer.rs (hunk 2 above)
Comment (notmandatory) [reply examples/psbt_signer.rs]: Good point, I removed it, was only there from the test code I was originally asked to look at.

File: examples/psbt_signer.rs (hunk 4 above)
Comment (notmandatory) [reply examples/psbt_signer.rs]: Another good idea ! When wallet balance < required I show the user the URL to uo1.net testnet faucet pre-prepopulated with address to send to (couldn't prepopulate the amount 🤷 ), when balance unconfirmed I show an message to wait, and when adequate balance confirmed voilà PSBT signed and broadcast.

File: examples/psbt_signer.rs (hunk 3 above)
Comment (notmandatory) [reply examples/psbt_signer.rs]: Done! do the new messages I added make sense?

**Comment by notmandatory** [comment]
I incorporated the suggested changes and also added a list of any unconfirmed tx if the wallet balance is less than the desired 10k sats. Ready to re-review! 🙏

**Review by danielabrozzoni — APPROVED** [review]
Looking great!

ACK fa998de4b1849adcf43351dbec05b3de43e6694e

File: examples/psbt_signer.rs [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+            );
+        }
+    } else {
+        println!("Creating a PSBT sending 9800 SATs plus fee to the u01.net testnet faucet return address 'tb1ql7w62elx9ucw4pj5lgw4l028hmuw80sndtntxt'.");
+        let return_address = Address::from_str("tb1ql7w62elx9ucw4pj5lgw4l028hmuw80sndtntxt")?;
+        let mut builder = watch_only_wallet.build_tx();
+        builder
+            .add_recipient(return_address.script_pubkey(), 9_800)
+            .enable_rbf()
+            .fee_rate(FeeRate::from_sat_per_vb(1.0));
```
Comment (afilini) [inline examples/psbt_signer.rs:82]: I would use `.drain_wallet().drain_to(return_address.script_pubkey())` to make sure a change is not created. It's also a chance to show that BDK can do more than "just send x to this address".

**Comment by afilini** [comment]
I was also about to suggest using the `descriptor!()` macro but decided not to because I think it could be a bit overwhelming at first. Most users will probably have a string for their descriptor and this would look more familiar to them.

File: examples/psbt_signer.rs (hunk 5 above)
Comment (notmandatory) [reply examples/psbt_signer.rs:82]: That's a good idea but I do like having some coin selection and change going on, which I think will be how most new projects will be doing rather than sweeping all funds.

Comment (danielabrozzoni) [reply examples/psbt_signer.rs:82]: Now that I think about it, the problem of receiving 10_000 and sending 9_800 is that you create a small change (~150 sats?), which will be uneconomical to spend (the fees for spending it are greater than the utxo value) - which BDK's coin selection won't pick, unless it's explicitly instructed to do so.

What this means is: after running this example many times, we might have the wallet full of these rather small utxos which we can't spend, and users will start to see some weird errors: for example, they might see that the balance is 10_000 (~70 utxos, ~150 each), but the coin selection will fail saying that there are not enough funds.

We can prevent this either by sweeping all funds every time, or by asking users to fill the wallet with 10_000 sats, but then spending only a smaller number, like 5_000

Comment (danielabrozzoni) [reply examples/psbt_signer.rs:82]: Oh, nevermind, the CS is smart enough that it picks 200 sats as fee, leaving no residue in the wallet :)

Comment (notmandatory) [reply examples/psbt_signer.rs:82]: Yes I guess also in my testing it used the change for fee and left an empty wallet balance. Yea for BDK coin selection! So we think it's ok to do it this way and not use `drain_to` to ensure there's no change?

Comment (danielabrozzoni) [reply examples/psbt_signer.rs:82]: I think it's fine :)

Comment (afilini) [reply examples/psbt_signer.rs:82]: Yeah the end result should be the same because the leftover is lower than the dust threshold. I can go ahead and merge it like this, my suggestion was based on what I would have done personally, but it also looks good like this.

**Review by afilini — APPROVED** [review]
ACK fa998de4b1849adcf43351dbec05b3de43e6694e
